pkg/vulnsrc/vulnerability/vulnerability.go

```diff
@@ -15,15 +15,38 @@ var (
 		RubySec, RustSec, PhpSecurityAdvisories, NodejsSecurityWg, PythonSafetyDB}
 )
 
-func GetDetail(vulnID string) (types.Severity, string, string, []string) {
-	details, err := db.Config{}.GetVulnerabilityDetail(vulnID)
+// TODO: Until we can dependency inject, we need to monkey patch sadly
+var (
+	getVulnerabilityDetailFunc = db.Config{}.GetVulnerabilityDetail
+)
+
+func GetDetail(vulnID string) (types.Severity, types.VendorSeverity, string, string, []string) {
+	details, err := getVulnerabilityDetailFunc(vulnID)
 	if err != nil {
 		log.Println(err)
-		return types.SeverityUnknown, "", "", nil
+		return types.SeverityUnknown, nil, "", "", nil
 	} else if len(details) == 0 {
-		return types.SeverityUnknown, "", "", nil
+		return types.SeverityUnknown, nil, "", "", nil
+	}
+
+	return getSeverity(details), getVendorSeverity(details), getTitle(details), getDescription(details), getReferences(details)
+}
+
+func getVendorSeverity(details map[string]types.VulnerabilityDetail) types.VendorSeverity {
+	vs := make(types.VendorSeverity)
+	for vendor, detail := range details {
+		switch {
+		case detail.Severity != types.SeverityUnknown:
+			vs[vendor] = detail.Severity
+		case detail.SeverityV3 != types.SeverityUnknown:
+			vs[vendor] = detail.SeverityV3
+		case detail.CvssScore != 0:
+			vs[vendor] = scoreToSeverity(detail.CvssScore)
+		case detail.CvssScoreV3 != 0:
+			vs[vendor] = scoreToSeverity(detail.CvssScoreV3)
```

Comment:
I'm wondering why you are using different logic. Is there any reason?
https://github.com/aquasecurity/trivy-db/blob/c47c05449be94fe880be11e46c0f6aed18631e26/pkg/vulnsrc/vulnerability/vulnerability.go#L57-L60

Reply:
Actually good catch! This is a bug. The score can be negative as we have seen here https://github.com/nodejs/security-wg/pull/91/files

So this should strictly be `> 0`. I have added a test and fixed here: https://github.com/aquasecurity/trivy-db/pull/32/commits/cb0286151707122fe537bfc7c0c6db2dae979c75

Reply:
Yes, we have a negative score.